src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurStandardAccountingExtractValidationServiceImpl.java

```diff
@@ -0,0 +1,90 @@
+package edu.cornell.kfs.concur.batch.service.impl;
+
+import java.sql.Date;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.log4j.Logger;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+
+import edu.cornell.kfs.concur.ConcurConstants;
+import edu.cornell.kfs.concur.batch.businessobject.ConcurStandardAccountingExtractDetailLine;
+import edu.cornell.kfs.concur.batch.businessobject.ConcurStandardAccountingExtractFile;
+import edu.cornell.kfs.concur.batch.service.ConcurStandardAccountingExtractValidationService;
+
+public class ConcurStandardAccountingExtractValidationServiceImpl implements ConcurStandardAccountingExtractValidationService {
+    private static final Logger LOG = Logger.getLogger(ConcurStandardAccountingExtractValidationServiceImpl.class);
+    
+    @Override
+    public boolean validateConcurStandardAccountExtractFile(ConcurStandardAccountingExtractFile concurStandardAccountingExtractFile) {
+        boolean valid = validateDetailCount(concurStandardAccountingExtractFile);
+        valid = validateAmounts(concurStandardAccountingExtractFile) && valid;
+        valid = validateDate(concurStandardAccountingExtractFile.getBatchDate()) && valid;
+        if (LOG.isDebugEnabled() && valid) {
+            LOG.debug("validateConcurStandardAccountExtractFile, passed file level validation, the record counts, batch date, and journal totals are all correct.");
+        }
+        return valid;
+    }
+
+    @Override
+    public boolean validateDetailCount(ConcurStandardAccountingExtractFile concurStandardAccountingExtractFile) {
+        Integer numberOfDetailsInHeader = concurStandardAccountingExtractFile.getRecordCount();
+        int actualNumberOfDetails = concurStandardAccountingExtractFile.getConcurStandardAccountingExtractDetailLines().size();
+        
+        boolean valid = numberOfDetailsInHeader.intValue() == actualNumberOfDetails;
+        
+        if (valid) {
+            LOG.debug("validateDetailCount, Number of detail lines is what we expected: " + actualNumberOfDetails);
+        } else {
+            LOG.error("validateDetailCount, The header said there were " + numberOfDetailsInHeader + 
+                    " the but the actual number of details was " + actualNumberOfDetails);
+        }
+        
+        return valid;
+    }
+
+    @Override
+    public boolean validateAmounts(ConcurStandardAccountingExtractFile concurStandardAccountingExtractFile) {
```

Comment:
this should be called something like: validateAmountsAndDebitCreditCode since it also checks the debit credit code